pkg/cmd/pulumi/plugin/plugin_prune.go

```diff
@@ -0,0 +1,288 @@
+		&latestOnly, "latest-only", "l", false,
+		"Keep only the absolute latest version of each plugin (ignoring major version differences)")
+
+	return cmd
+}
+
+// Run executes the plugin prune command
+func (cmd *pluginPruneCmd) Run() error {
+	opts := display.Options{
+		Color: cmdutil.GetGlobalColorization(),
+	}
+
+	// Get all plugins
+	plugins, err := cmd.getPluginsWithMetadata()
+	if err != nil {
+		return fmt.Errorf("loading plugins: %w", err)
+	}
+
+	if len(plugins) == 0 {
+		cmd.diag.Infof(
+			diag.Message("", "no plugins found to prune"))
+		return nil
+	}
+
+	// Group plugins by kind, name, and major version
+	groups := make(map[string][]workspace.PluginInfo)
+	for _, plugin := range plugins {
+		// Skip bundled plugins - we don't want to mess with these
+		if workspace.IsPluginBundled(plugin.Kind, plugin.Name) {
+			continue
+		}
+
+		// Create a key that includes kind, name, and major version (if available)
+		var key string
+		if plugin.Version != nil {
+			if cmd.latestOnly {
+				// When using latestOnly, only group by kind and name
+				key = fmt.Sprintf("%s|%s", plugin.Kind, plugin.Name)
+			} else {
+				// Group by kind, name, and major version
+				key = fmt.Sprintf("%s|%s|v%d", plugin.Kind, plugin.Name, plugin.Version.Major)
+			}
+		} else {
+			// If no version, just use kind and name
+			key = fmt.Sprintf("%s|%s", plugin.Kind, plugin.Name)
+		}
+
+		groups[key] = append(groups[key], plugin)
+	}
+
+	// For each group, identify plugins to remove (all but the latest version)
+	toRemove := make([]workspace.PluginInfo, 0, len(plugins))
+	toKeep := make([]workspace.PluginInfo, 0, len(plugins))
+	var totalSizeRemoved uint64
+
+	for _, group := range groups {
+		if len(group) <= 1 {
+			// Only one version, keep it
+			toKeep = append(toKeep, group[0])
+			continue
+		}
+
+		// Sort plugins by version in descending order (newer versions first).
+		// Plugins with nil versions are treated as lower priority and are sorted to
+		// the end of the list. This ensures plugins with defined versions are preferred
+		// when determining which version to keep.
+		sort.Slice(group, func(i, j int) bool {
+			// If either version is nil, keep the one with a version
+			if group[i].Version == nil {
+				return false
+			}
+			if group[j].Version == nil {
+				return true
+			}
+			// Otherwise sort by version (newer versions first)
+			return group[i].Version.GT(*group[j].Version)
+		})
+
+		// Keep only the latest version in each group
+		toKeep = append(toKeep, group[0])
+
+		// Remove the rest
+		for i := 1; i < len(group); i++ {
+			toRemove = append(toRemove, group[i])
+			totalSizeRemoved += group[i].Size
+		}
+	}
+
+	if len(toRemove) == 0 {
+		cmd.diag.Infof(
+			diag.Message("", "no plugins found to prune"))
+		return nil
+	}
+
+	// Confirm that the user wants to do this (unless --yes was passed)
+	if !cmd.dryRun && !cmd.yes {
+		var suffix string
+		if len(toRemove) != 1 {
+			suffix = "s"
+		}
+		fmt.Print(
```

Comment:
Let's use `cmd.Stdout` to keep this testable.
```suggestion
		fmt.Fprint(cmd.Stdout,
```